extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -118,15 +126,50 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
 
   /**
    * Triggers template update if opening without active product data.
-   * This happens when the "Shop Now" CTA is clicked.
+   * This happens either when the "Shop Now" CTA is clicked or when a
+   * shopping tag is clicked.
    * @param {boolean} isOpen
    * @private
    */
   onAttachmentStateUpdate_(isOpen) {
     if (!this.isOnActivePage_() || !isOpen) {
       return;
     }
+
+    const shoppingItemClicked =
+      document.activeElement.tagName === 'AMP-STORY-SHOPPING-TAG'
+        ? document.activeElement.tagName
+        : 'AMP-STORY-SHOPPING-ATTACHMENT';
     const shoppingData = this.storeService_.get(StateProperty.SHOPPING_DATA);
+    const pageId = this.pageEl_.id;
+    const activeProuductData = Object.values(shoppingData[pageId]);
+
+    if (activeProuductData.length === 1) {
+      this.storeService_.dispatch(Action.ADD_SHOPPING_DATA, {
+        'activeProductData': activeProuductData[0],
+      });
+    }
+
+    if (shoppingItemClicked === 'AMP-STORY-SHOPPING-ATTACHMENT') {
+      if (activeProuductData.length === 1) {
+        this.variableService_.onVariableUpdate(
+          AnalyticsVariable.STORY_SHOPPING_PRODUCT_ID,
+          activeProuductData[0].productId
+        );
+        this.analyticsService_.triggerEvent(
+          StoryAnalyticsEvent.SHOPPING_PDP_VIEW
+        );
+      } else if (shoppingData?.activeProductData?.productId === undefined) {
+        this.variableService_.onVariableUpdate(
+          AnalyticsVariable.STORY_PAGE_ID,
+          pageId
+        );
+        this.analyticsService_.triggerEvent(
+          StoryAnalyticsEvent.SHOPPING_PLP_VIEW
+        );
+      }
+    }
```

Comment:
Shopping data for the template is handled in `updateTemplate_` and the value is stored within productForPdp. If we do the logic there it could be as compact as:
<img width="500" alt="Screen Shot 2022-03-31 at 3 17 17 PM" src="https://user-images.githubusercontent.com/3860311/161142184-26166270-c38a-46e5-8efc-78eaca348a02.png">

We could then remove the analytics code in `onPlpCardClick_`.

Reply:
This is a good idea, but it doesn't take into account the case when there is a plp view with multiple products on the page. I've added some logic to take care of that case.

```

   //pdp view
    if (productForPdp) {
      this.variableService_.onVariableUpdate(
        AnalyticsVariable.STORY_SHOPPING_PRODUCT_ID,
        productForPdp.productId
      );
    }
    //plp view with more than one product on the page
    else {
      const pageId = this.pageEl_.id;
      this.variableService_.onVariableUpdate(
        AnalyticsVariable.STORY_PAGE_ID,
        pageId
      );
    }

```